hack/generator/pkg/astmodel/type_name.go

```diff
@@ -66,6 +66,26 @@ func (typeName TypeName) AsType(codeGenerationContext *CodeGenerationContext) ds
 	return dst.NewIdent(typeName.name)
 }
 
+// AsZero renders an expression for the "zero" value of the type.
+// The exact thing we need to generate depends on the actual type we reference
+func (typeName TypeName) AsZero(types Types, ctx *CodeGenerationContext) dst.Expr {
+	actualType, err := types.FullyResolve(typeName)
+	if err != nil {
+		// This should never happen
+		panic(err)
+	}
+
+	if _, isObject := AsObjectType(actualType); isObject {
+		// We reference an object type, so our zero value is an empty struct
+		return &dst.BasicLit{
+			Value: fmt.Sprintf("%s{}", typeName.Name()),
```

Comment:
Ahhhh, is this why `ObjectType.AsZero` panics, because you can't create the zero value of an unnamed struct?

Reply:
yeah agree, this makes sense now

Reply:
I've added a comment to `ObjectType.AsZero()` to make this more clear.